core/src/main/scala/com/gu/AppIdentity.scala

```diff
@@ -13,30 +15,28 @@ import scala.util.{Failure, Success, Try}
 sealed trait AppIdentity
 
 case class AwsIdentity(
-                        app: String,
-                        stack: String,
-                        stage: String,
-                        region: String
-                      ) extends AppIdentity
+  app: String,
+  stack: String,
+  stage: String,
+  region: String
+) extends AppIdentity
 
-case class DevIdentity(
-                        app: String
-                      ) extends AppIdentity
+case class DevIdentity(app: String) extends AppIdentity
 
 object AppIdentity {
   private val logger = LoggerFactory.getLogger(this.getClass)
 
-  private def safeAwsOperation[A](errorMessage: => String)(operation: => A): Option[A] = Try(operation) match {
-    case Success(value) => Option(value)
-    case Failure(e) =>
-      logger.error(errorMessage, e)
-      None
-  }
+  private def safeAwsOperation[A](errorMessage: => String)(operation: => A): Try[A] =
+    Try(operation).recoverWith {
```

Comment:
I was hoping for a kind of `failureForEach` or `failureMap` but the closest I could find was `recoverWith`

Reply:
that's a shame! The only other thing I can think of is:
```
    val result = Try(operation)
    result.failed.foreach(e => logger.error(errorMessage, e))
    result
```

Reply:
I didn't know about that! Very strange operation, but it works.
I will use that instead because it is less powerful
`Inverts this Try. If this is a Failure, returns its exception wrapped in a Success. If this is a Success, returns a Failure containing an UnsupportedOperationException.`